libminifi/include/utils/ChecksumCalculator.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <utility>
+
+#include "utils/OptionalUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+class ChecksumCalculator {
+ public:
+  static constexpr const char* CHECKSUM_TYPE = "SHA256";
+  static constexpr size_t LENGTH_OF_HASH_IN_BYTES = 32;
+
+  void setFileLocation(const std::string& file_location);
+  std::string getFileName() const;
+  std::string getChecksum();
+  void invalidateChecksum();
+
+ private:
+  static std::string computeChecksum(const std::string& file_location);
+
+  utils::optional<std::string> file_location_;
```

Comment:
What's the purpose of optionals here?
Maybe I miss something trivial, but I would expect checking the path before making the decision to compute the checksum of the given file.

Reply:
`Properties` and `FlowConfiguration` both contain a default-constructed `ChecksumCalculator`, but initially these are empty, so all these optionals are `nullopt`.  Setting the file location and computing the checksum only happens if the user enables it by adding `ConfigurationChecksums` to `nifi.c2.root.classes`.

Reply:
Okay, that makes perfect sense, thanks!